plugins/module_utils/data_set.py

```diff
@@ -919,7 +919,7 @@ def _build_zoau_args(**kwargs):
                 secondary += space_type
 
         type = kwargs.get("type")
-        if type and type == "ZFS":
+        if type and type.upper() == "ZFS":
             type = "LDS"
```

Comment:
shouldn't this be 'type="lds""?

Reply:
This particular call to ZOAU requires args to be uppercase, hence the call to .upper(). I originally changed it to lowercase and only found out in testing that it broke this code path.